core/server/api/authentication.js

```diff
@@ -42,7 +43,7 @@ function assertSetupCompleted(status) {
                 notCompleted = i18n.t('errors.api.authentication.setupMustBeCompleted');
 
             function throwReason(reason) {
-                throw new errors.NoPermissionError(reason);
+                throw new errors.NoPermissionError({message: reason});
```

Comment:
I don't understand the reason for these changes. There are a lot of them and they make this PR incredibly difficult to review. If we are going to change our error signature to be different, I think this should probably be a  separate PR?